src/main/java/org/moltimate/moltimatebackend/service/LigandService.java

```diff
@@ -0,0 +1,104 @@
+import org.moltimate.moltimatebackend.exception.DockingJobFailedException;
+import org.moltimate.moltimatebackend.exception.InvalidFileException;
+import org.moltimate.moltimatebackend.model.ActiveSite;
+import org.moltimate.moltimatebackend.model.Motif;
+import org.moltimate.moltimatebackend.model.Residue;
+import org.moltimate.moltimatebackend.model.ResidueQuerySet;
+import org.moltimate.moltimatebackend.repository.MotifRepository;
+import org.moltimate.moltimatebackend.repository.ResidueQuerySetRepository;
+import org.moltimate.moltimatebackend.util.ActiveSiteUtils;
+import org.moltimate.moltimatebackend.util.DockingUtils;
+import org.moltimate.moltimatebackend.util.MotifUtils;
+import org.moltimate.moltimatebackend.util.ProteinUtils;
+import org.moltimate.moltimatebackend.util.StructureUtils;
+import org.moltimate.moltimatebackend.util.DockingUtils.InMemoryMultipartFile;
+import org.moltimate.moltimatebackend.validation.EcNumberValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class LigandService {
+
+    @Autowired
+    private MotifRepository motifRepository;
+
+    /**
+     * @param ecNumber EC number to find associated Ligands for
+     * @return List of Ligands associated with Proteins of this EC
+     */
+    public List<RCSBLigand> getByEcNumber(String ecNumber) {
+        if (ecNumber == null || EcNumber.UNKNOWN.equals(ecNumber)) {
+            return Collections.emptyList();
+        }
+        EcNumberValidator.validate(ecNumber);
+        List<String> pdbIds = motifRepository
+            .findByEcNumberEqualsOrEcNumberStartingWith(
+                EcNumber.UNKNOWN, ecNumber, PageRequest.of(0, 320))
+            .map(motif -> motif.getPdbId()).getContent();
+        
+        log.info("Retrieving Ligands associated with PDB IDs {}", pdbIds);
+
+        List<RCSBLigand> matchingLigands = RCSBLigandsFactory.getFromPdbIds(pdbIds).stream()
+            .flatMap(ligands -> ligands.getLigands().stream())
+            .collect(Collectors.groupingBy(l -> l.getName()))
+            .values()
+            .stream()
+            .flatMap(group -> group.stream().limit(1))
+            .collect(Collectors.toList());
+
+        return matchingLigands;
+    }
+
+    /**
+     * Fetch the Ligand File needed for the supplied Docking Request
+     * @param request
+     */
+    public static InMemoryMultipartFile fetchLigand(DockingRequest request) {
```

Comment:
This should probably be just a MultipartFile here, not an InMemoryMultipartFile.